pkg/mountinfo/mountinfo_linux.go

```diff
@@ -0,0 +1,128 @@
+// +build linux
+
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mountinfo
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	// Number of fields per line in /proc/mounts as per the fstab man page.
+	expectedNumFieldsPerLine = 6
+	// Location of the mount file to use
+	procMountsPath = "/proc/mounts"
+)
+
+// internal function implements same functionality as list of all mounted filesystems.
+func listProcMounts(procMountsPath, prefix string) ([]mountInfo, error) {
+	mounts, err := readProcMounts(procMountsPath)
+	if err != nil {
+		return nil, err
+	}
+	var filteredMounts []mountInfo
+	for _, mount := range mounts {
+		// Add a separator to indicate that this is a proper directory/device.
+		// This is to avoid a situation where prefix is '/tmp/fsmount'
+		// and mount path is /tmp/fs. In-fact we need to check for `/tmp/fs/`
+		if strings.HasPrefix(strings.TrimSuffix(mount.Path, "/")+"/",
```

Comment:
path.Abs() is still needed here, otherwise sometimes a mounted dir inside the bucket won't be detected,

```
$ mkdir -b /tmp/fs/bucket
$ sudo mount /dev/your_disk /tmp/fs/bucket
$ cd /tmp/fs/
$ minio server .
```